tests/scripts/thread-cert/Cert_5_3_06_RouterIdMask.py

```diff
@@ -89,64 +113,99 @@ def test(self):
-        # mins.
-        routing_cost = 1
-        for i in range(0, 24):
-            self.simulator.go(30)
-            print("%ss" % ((i + 1) * 30))
-
-            leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-            msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT, False)
-            if msg is None:
-                continue
-
-            self.assertTrue(command.check_id_set(msg, router2_id))
-
-            routing_cost = command.get_routing_cost(msg, router2_id)
-            if routing_cost == 0:
-                break
-        self.assertTrue(routing_cost == 0)
-
-        self.simulator.go(config.INFINITE_COST_TIMEOUT + config.MAX_ADVERTISEMENT_INTERVAL)
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
-        self.assertFalse(command.check_id_set(msg, router2_id))
+
+        self.simulator.go(720)
 
         # 5
-        # Flush the message queue to avoid possible impact on follow-up
-        # verification.
-        self.simulator.get_messages_sent_by(DUT_LEADER)
 
         self.nodes[ROUTER2].start()
         self.simulator.go(5)
         self.assertEqual(self.nodes[ROUTER2].get_state(), 'router')
 
         self.simulator.go(config.MAX_ADVERTISEMENT_INTERVAL)
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
 
         # 6
         self.nodes[ROUTER1].reset()
         self.nodes[ROUTER2].reset()
 
-        router1_id = self.nodes[ROUTER1].get_router_id()
-        router2_id = self.nodes[ROUTER2].get_router_id()
-
-        self.simulator.go(config.MAX_NEIGHBOR_AGE + config.MAX_ADVERTISEMENT_INTERVAL)
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
-        self.assertEqual(command.get_routing_cost(msg, router1_id), 0)
-
-        self.simulator.go(config.INFINITE_COST_TIMEOUT + config.MAX_ADVERTISEMENT_INTERVAL)
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
-        self.assertFalse(command.check_id_set(msg, router1_id))
-        self.assertFalse(command.check_id_set(msg, router2_id))
+        self.simulator.go(720)
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+
+        # Step 1: Ensure topology is formed correctly
+        pv.verify_attached('ROUTER_1', 'LEADER')
+        pv.verify_attached('ROUTER_2', 'ROUTER_1')
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,2,1} == set(p.mle.tlv.route64.cost)).\
```

Comment:
Is this ordered by Router ID? If it is, we can compare using lists. 
We can get the router IDs of each device so that we can order these costs.

Reply:
yes, The routing table entry is ordered by the router id

Reply:
add router id checks instead of check orders